### PR TITLE
Merge feature/merchant index/story 41 into master

### PR DESCRIPTION
**Pull request by Capleugh**
**Functionality:**
- Create model method to activate all items
- Moved model method calls into change_merchant_status method

**Testing:**
- 100%
- Working in local development 

**Related Issues:**
#91

### CODE REVIEW

File: app/controllers/admin/merchants_controller.rb [hunk 1]
```diff
@@ -9,21 +9,21 @@ def show
...
       if merchant.enabled?
         merchant.update(enabled?: false)
+        merchant.items.deactivate_all_items
 
         flash[:success] = "#{merchant.name} is now disabled."
       else
         merchant.update(enabled?: true)
+        merchant.items.activate_all_items
 
         flash[:success] = "#{merchant.name} is now enabled."
```
Comment (mintona) [inline app/controllers/admin/merchants_controller.rb:28]: This looks super clean.

File: app/views/admin/items/index.html.erb [hunk 2]
```diff
@@ -11,10 +11,8 @@
         <p>Inventory: <%= item.inventory %> </p>
       <% if item.active? %>
         <p>Status: Active</p>
-        <%# <%= button_to 'Deactivate', "/merchants/#{@merchant.id}/items/#{item.id}?status=deactivate", method: :patch %> %>
       <% else %>
         <p>Status: Inactive</p>
-        <%# <%= button_to 'Activate', "/admin/merchants/#{@merchant.id}/items/#{item.id}?status=activate", method: :patch %> %>
```
Comment (mintona) [inline app/views/admin/items/index.html.erb:17]: I will try to get clarity on this for going forward. Good to go like this per story.

File: spec/models/item_spec.rb [hunk 3]
```diff
@@ -152,5 +152,18 @@
...
       expect(Item.find(item_5.id).active?).to eq(true)
     end
+
+    it "activate_all_items" do
+      dog_shop = create(:merchant, enabled?: false)
+      item_1 = create(:item, active?: false, merchant: dog_shop)
+      item_2 = create(:item, active?: false, merchant: dog_shop)
+      item_3 = create(:item, active?: false, merchant: dog_shop)
+
+      dog_shop.items.activate_all_items
```
Comment (mintona) [inline spec/models/item_spec.rb]: I thought about it some more and I think these tests can safely use `Items.activate_all_items` with items from just one shop, as the point of this method is to change any items it's called on to activate (or in the other case, deactive) items. It's job is not to find the items it should be working on, that's our job to make that collection available.

**Review by mintona — APPROVED** [review]
Nice, Carleigh! Looks super clean. Just left a comment on the testing for the item model method we were discussing today.

File: spec/models/item_spec.rb (hunk 3 above)
Comment (Capleugh) [reply spec/models/item_spec.rb]: Oooh, yes. I like that reasoning. I'll go change it.

**Review by hale4029 — APPROVED** [review]
Nice, looks good.

**Pull request merged**